File: test/c/intel_x64/leaf_01.c
#include <pal/cpuid/leaf_01_eax.h>
#include <pal/cpuid/leaf_01_ebx.h>
#include <pal/cpuid/leaf_01_ecx.h>
#include <pal/cpuid/leaf_01_edx.h>

void test_leaf_01_compile(void)
{
    // Register accessors
    uint32_t eax = pal_leaf_01_eax_get();
    uint32_t ebx = pal_leaf_01_ebx_get();
    uint32_t ecx = pal_leaf_01_ecx_get();
    uint32_t edx = pal_leaf_01_edx_get();

    // Field accessors
    pal_leaf_01_eax_model_get();
    pal_leaf_01_eax_model_get_from_value(eax);

    // Printers
    pal_leaf_01_eax_dump();
    pal_leaf_01_eax_dump_from_value(eax);
    pal_leaf_01_eax_model_dump();
    pal_leaf_01_eax_model_dump_from_value(eax);
}

File: test/rust/intel_x64/src/leaf_01.rs
use pal;

pub fn test_leaf_01_compile()
{
    let eax = pal::cpuid::leaf_01_eax::get();
    let _ebx = pal::cpuid::leaf_01_ebx::get();
    let _ecx = pal::cpuid::leaf_01_ecx::get();
    let _edx = pal::cpuid::leaf_01_edx::get();

    let _value = pal::cpuid::leaf_01_eax::get_model();
    let _value2 = pal::cpuid::leaf_01_eax::get_model_from_value(eax);

    pal::cpuid::leaf_01_eax::print();
    pal::cpuid::leaf_01_eax::print_from_value(eax);
    pal::cpuid::leaf_01_eax::print_model();
    pal::cpuid::leaf_01_eax::print_model_from_value(eax);
}
